### PR TITLE
[PFT-1796] Add changes for the snx subgraph code and order

### PR DESCRIPTION
**Pull request by Akhilleshgoswami**

### CODE REVIEW

File: src/common/subgraphMapper.ts [hunk 1]
```diff
@@ -1,62 +1,40 @@
...
-  VaultPosition,
+  Wallet,
 } from '../interfaces/subgraphTypes';
 
 ////////////////////////////////////////////////////////////////
 //////////////////////    ACCOUNT   ////////////////////////////
 ////////////////////////////////////////////////////////////////
 
-export const mapSubgraphResponseToAccountInterface = (response: any): Account | undefined => {
+export const mapSubgraphResponseToAccountInterface = (response: any): Wallet | undefined => {
```
Comment (sudeepb02) [inline src/common/subgraphMapper.ts]: Rename all instances of previous Account type to `Wallet` type

File: src/common/subgraphMapper.ts [hunk 2]
```diff
@@ -1,62 +1,40 @@
...
-      totalStaked: response.totalStaked,
-      esPRFBalance: response.esPRFBalance,
     };
   } catch (error) {
     console.log('Error while mapping data', error);
     throw error;
   }
 };
 
-export const mapAccountsArrayToInterface = (response: any): Account[] | undefined => {
```
Comment (sudeepb02) [inline src/common/subgraphMapper.ts:49]: This function will still be needed for mapping when multiple wallets are fetched in a single query

File: src/interfaces/sdkTypes.ts [hunk 3]
```diff
@@ -55,4 +55,105 @@ export interface LeaderboardUserData {
...
   totalRealizedPnlPositions: Decimal;
   totalAccruedBorrowingFeesInUsd: Decimal;
-}
\ No newline at end of file
+} 
+export type Market = {
+  id: string;
+  marketName: string;
+  marketSymbol: string;
+  size: string;
```
Comment (sudeepb02) [inline src/interfaces/sdkTypes.ts:63]: Add fields for formatted size. Please check with @iamamitkumar0512 to add other fields that we need to calculate to determine the skew and OI of markets for longs and shorts

File: src/interfaces/sdkTypes.ts [hunk 4]
```diff
@@ -55,4 +55,105 @@ export interface LeaderboardUserData {
...
+  market?: Market;
+  user?: Wallet;
+  isLimitOrder: boolean;
+  expectedPrice: string;
+  expectedPriceTime?: string;
+  settlementTime?: string;
+  deadline: string;
+  trackingCode?: string;
+  deltaCollateral: string;
+  deltaSize: string;
```
Comment (sudeepb02) [inline src/interfaces/sdkTypes.ts:104]: Add field for formatted deltaSize

File: src/subgraph/markets/subgraphQueries.ts [hunk 5]
```diff
@@ -73,67 +24,18 @@ export const fetchAllMarketsDataQuery = gql`
 export const fetchMarketByIdQuery = (marketId: string) => gql`
   {
     market(id: "${marketId.toLowerCase()}") {
```
Comment (sudeepb02) [inline src/subgraph/markets/subgraphQueries.ts:26]: Have we tested this? The market id in the latest snx subgraph is a number (even though IDs are stored as strings). Please check the subgraph.yaml and the returned data.
Also update the test cases

**Review by sudeepb02 — CHANGES_REQUESTED** [review]

File: src/interfaces/sdkTypes.ts (hunk 4 above)
Comment (Akhilleshgoswami) [reply src/interfaces/sdkTypes.ts:104]: we are having deltaSize in USD will it not work ??

Comment (sudeepb02) [reply src/interfaces/sdkTypes.ts:104]: That will be a snapshot of the size in the past based on the price of the token at that timestamp. And here we are adding the formatted delta size in token itself. Both are different

File: src/interfaces/subgraphTypes.ts [hunk 6]
```diff
@@ -164,12 +170,11 @@ export interface Order {
   /** True if it is a limit order */
   isLimitOrder: boolean;
   /** Acceptable price */
```
Comment (sudeepb02) [inline src/interfaces/subgraphTypes.ts]: The comments are not aligned here

File: test/common/constants.ts [hunk 7]
```diff
@@ -10,7 +10,7 @@ export const TEST_POSITION_ID3 = '0xe85f118865dfdae84a450360d6e0c3e601218f312a0a
 export const TEST_POSITION_ID4 = '0xa1f496b9eecc3e711a3cc741ba3d34abfdebdb73e6fb3ff5ef187196350177dc';
 export const TEST_OPEN_POSITION = '0xa9a3ba329fce666f22682ad894a0031c2cb44d62451a1a52b67df0857175b547';
 
-export const TEST_USER_ID1 = '0x850cf0ff7b0bf01255ddfce1d3dc4e1cbfeaf5af';
+export const TEST_USER_ID1 = 'PERP-170141183460469231731687303715884105729';
```
Comment (sudeepb02) [inline test/common/constants.ts]: User ID is incorrect? Is this just for testing purposes?

File: test/subgraph-tests/accounts.test.ts [hunk 8]
```diff
@@ -1,5 +1,5 @@
 import { getParifiSdkInstanceForTesting } from '..';
-import { TEST_USER_ID1, TEST_USER_ID2, TEST_USER_ID3 } from '../common/constants';
+import { TEST_USER_ID1 as snxAccountId, TEST_USER_ID2, TEST_USER_ID3 } from '../common/constants';
```
Comment (sudeepb02) [inline test/subgraph-tests/accounts.test.ts]: Incorrect assumption. SNX account id is a bigint. The ID of the subgraph entity vs. the ID of the snxAccount both are different. Please check subgraph schema and reach out to me for explanation

**Review by sudeepb02 — CHANGES_REQUESTED** [review]

File: test/common/constants.ts (hunk 7 above)
Comment (Akhilleshgoswami) [reply test/common/constants.ts]: yes

Comment (Akhilleshgoswami) [reply test/common/constants.ts]: it's the perp_Account ID

File: package.json [hunk 9]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@parifi/sdk",
-  "version": "1.0.12",
+  "version": "1.0.31-dev",
```
Comment (sudeepb02) [inline package.json]: Update the sdk version to use `2.0.0` to indicate major changes with the switch to Synthetix

File: src/common/subgraphMapper.ts [hunk 10]
```diff
@@ -1,63 +1,34 @@
...
-      totalRealizedPnlVaults: response.totalRealizedPnlVaults,
+      openPositionCount: response.openPositionCount,
       countProfitablePositions: response.countProfitablePositions,
       countLossPositions: response.countLossPositions,
       countLiquidatedPositions: response.countLiquidatedPositions,
       totalVolumeInUsd: response.totalVolumeInUsd,
       totalVolumeInUsdLongs: response.totalVolumeInUsdLongs,
       totalVolumeInUsdShorts: response.totalVolumeInUsdShorts,
       totalAccruedBorrowingFeesInUsd: response.totalAccruedBorrowingFeesInUsd,
-      totalStaked: response.totalStaked,
```
Comment (sudeepb02) [inline src/common/subgraphMapper.ts:40]: We still need these subgraph fields for the staking page, these are still relevant and applicable

File: src/common/subgraphMapper.ts [hunk 11]
```diff
@@ -137,48 +85,65 @@ export const mapMarketsArrayToInterface = (response: any): Market[] | undefined
...
+      settledTimestampISO: orderResponse.settledTimestampISO,
+      executionPrice: orderResponse.executionPrice,
+      formattedExecutionPrice: formatEther(orderResponse.executionPrice),
+      expectedPrice: orderResponse.acceptablePrice,
+      formateedExpectedPrice: formatEther(orderResponse.acceptablePrice),
+      settledBy: orderResponse.settledBy ? mapSubgraphResponseToWalletInterface(orderResponse.settledBy) : undefined,
+      positionId: orderResponse.position ? orderResponse.position.id : undefined,
+      formattedDeltaSize: formatEther(orderResponse.deltaSize),
+      depositCollateral: depositCollateral,
+      snxAccount: orderResponse.snxAccount.id,
```
Comment (sudeepb02) [inline src/common/subgraphMapper.ts]: This should be mapped to the snxAccountId and not the subgraph entity id

File: src/common/subgraphMapper.ts [hunk 12]
```diff
@@ -137,48 +85,65 @@ export const mapMarketsArrayToInterface = (response: any): Market[] | undefined
...
-export const mapOrdersArrayToInterface = (response: any): Order[] | undefined => {
+export const mapOrdersArrayToInterface = (
+  response: any,
+  collateralDepositResponse: Record<string, DepositCollateral[]>,
+): Order[] | undefined => {
   if (response === null) return undefined;
   try {
     return response.orders.map((order: Order) => {
-      return mapSingleOrderToInterface(order);
+      const depositedCollateral = collateralDepositResponse[order?.snxAccount?.id || ''];
```
Comment (sudeepb02) [inline src/common/subgraphMapper.ts:136]: Have we tested this?

File: src/common/subgraphMapper.ts [hunk 13]
```diff
@@ -190,50 +155,54 @@ export const mapOrdersArrayToInterface = (response: any): Order[] | undefined =>
...
       avgPrice: response.avgPrice,
-      avgPriceDec: response.avgPriceDec,
-      lastCumulativeFee: response.lastCumulativeFee,
+      formattedAvgPrice: formatEther(response.avgPrice),
       status: response.status,
+      snxAccount: response.snxAccount.id,
       txHash: response.txHash,
       liquidationTxHash: response.liquidationTxHash,
       closingPrice: response.closingPrice,
+      formattedClosingPrice: formatEther(response.closingPrice),
```
Comment (sudeepb02) [inline src/common/subgraphMapper.ts]: `formatEther` might dump if `undefined` is passed to it. Please test @Akhilleshgoswami @iamamitkumar0512

File: src/interfaces/sdkTypes.ts [hunk 14]
```diff
@@ -1,4 +1,6 @@
 import Decimal from 'decimal.js';
+import { OrderStatus, PositionStatus, PythData, SnxAccountType } from './subgraphTypes';
+import { S } from '@parifi/synthetix-sdk-ts/dist/index-BTlMb-Ja';
```
Comment (sudeepb02) [inline src/interfaces/sdkTypes.ts]: Remove this. No import should be done from the dist directory

File: src/interfaces/subgraphTypes.ts [hunk 15]
```diff
@@ -18,413 +18,236 @@ export enum OrderStatus {
...
+  interestRate:string
 }
 
+
 ////////////////////////////////////////////////////////////////
 //////////////////////    ORDERS    ////////////////////////////
 ////////////////////////////////////////////////////////////////
-
+// settlementReward, referralFees, partnerAddressts
 export interface Order {
```
Comment (sudeepb02) [inline src/interfaces/subgraphTypes.ts:161]: Mark all fields as optional for all responses in subgraph types

File: src/perps/positions.ts [hunk 16]
```diff
@@ -0,0 +1,61 @@
...
+import { formatEther } from 'ethers';
+
+export const getProfitOrLossInUsd = (
+  positionData: Position,
+  normalizedMarketPrice: Decimal,
+  marketDecimals: number = 18,
+): { totalProfitOrLoss: Decimal } => {
+  let profitOrLoss: Decimal;
+  const { avgPrice, positionSize, isLong } = positionData;
+  const positionAvgPrice = new Decimal(avgPrice).div(Decimal.pow(10, marketDecimals));
```
Comment (sudeepb02) [inline src/perps/positions.ts:12]: Add helper functions for size conversions

File: src/perps/positions.ts [hunk 17]
```diff
@@ -0,0 +1,61 @@
...
+  const positionSizeDecimal = new Decimal(Math.abs(Number(positionSize))).abs().div(Decimal.pow(10, marketDecimals));
+  if (isLong) {
+    // If long, profit when market price > avg price
+    if (normalizedMarketPrice.gt(positionAvgPrice)) {
+      profitOrLoss = normalizedMarketPrice.minus(positionAvgPrice);
+    } else {
+      profitOrLoss =  positionAvgPrice.minus(normalizedMarketPrice);
+      profitOrLoss.times(-1)
```
Comment (sudeepb02) [inline src/perps/positions.ts:20]: This logic is incorrect. Please test with sample values and fix the logic

File: src/perps/positions.ts [hunk 18]
```diff
@@ -0,0 +1,61 @@
...
+  if (!position || !position.depositCollateral?.[0]) {
+    return { positionLeverage: new Decimal(0) };
+  }
+
+  // Calculate collateral in USDC
+  const collateralUsed = position.depositCollateral[0].formattedDepositedAmount || 0;
+  const collateralInUSDC = new Decimal(collateralUsed).times(collateralPrice);
+
+  // Calculate position size in USDC
+  const positionSize = new Decimal(formatEther(BigInt(position.positionSize || 0)));
```
Comment (sudeepb02) [inline src/perps/positions.ts:50]: Use helper functions for conversions

File: src/relayers/pimlico/index.ts [hunk 19]
```diff
@@ -1,118 +0,0 @@
-import 'dotenv/config';
```
Comment (sudeepb02) [inline src/relayers/pimlico/index.ts]: @Akhilleshgoswami We still use Pimlico relayers. Please leave Pimlico logic as it is and remove the rest from relayers

File: src/subgraph/accounts/subgraphQueries.ts [hunk 20]
```diff
@@ -104,6 +104,14 @@ export const fetchTopAccountsByReferralFees = (count: number = 20, skip: number
     unclaimedReferralRewardsWeth
   }
 }`;
+export const fetchAccountByWalletAddress = (walletAddress: string) =>
```
Comment (sudeepb02) [inline src/subgraph/accounts/subgraphQueries.ts:107]: Wallet and account are both different terms in the subgraph. Please check the logic and avoid mixing up the terms. Use `Wallet` for a user address and use `snxAccount` for Synthetix account ids

File: src/subgraph/common/index.ts [hunk 21]
```diff
@@ -1,7 +1,7 @@
 import { Chain } from '@parifi/references';
 
 export const publicSubgraphEndpoints: { [key in Chain]: string } = {
-  [Chain.ARBITRUM_SEPOLIA]: 'https://api.thegraph.com/subgraphs/name/parifi/parifi-sepolia',
+  [Chain.ARBITRUM_SEPOLIA]: 'https://subgraph.satsuma-prod.com/ac10c1d41dcb/parifis-team--3804602/parifi-snx-sepolia/api',
```
Comment (sudeepb02) [inline src/subgraph/common/index.ts:4]: Use env variables for private endpoints. If public endpoints are not available, return empty string. 
Avoid using private endpoints in code directly

File: src/subgraph/markets/subgraphQueries.ts (hunk 5 above)
Comment (sudeepb02) [reply src/subgraph/markets/subgraphQueries.ts:26]: @Akhilleshgoswami Not sure if anything was changed here. Could you please explain?

File: src/subgraph/orders/index.ts [hunk 22]
```diff
@@ -25,8 +25,21 @@ export const getAllOrdersByUserAddress = async (
 ): Promise<Order[]> => {
   try {
     const subgraphResponse: any = await request(subgraphEndpoint, fetchOrdersByUserQuery(userAddress, count, skip));
+    if (!subgraphResponse) throw new Error('Error While Fechting All Order By user Address');
+    const accountIdArray = subgraphResponse?.orders?.map((order: Order) => {
+      return order?.snxAccount?.id;
```
Comment (sudeepb02) [inline src/subgraph/orders/index.ts:30]: Incorrect logic. Have we tested this? @Akhilleshgoswami

Comment (sudeepb02) [reply src/subgraph/orders/index.ts:30]: Please run the test cases and verify valid data response

File: src/subgraph/orders/index.ts [hunk 23]
```diff
@@ -25,8 +25,21 @@ export const getAllOrdersByUserAddress = async (
...
+    });
+    const collateralSubgraphResponse: any = await request(
+      subgraphEndpoint,
+      fectchCollateralForOrderUsingAccountId(accountIdArray),
```
Comment (sudeepb02) [inline src/subgraph/orders/index.ts]: typo here

File: src/subgraph/orders/subgraphQueries.ts [hunk 24]
```diff
@@ -190,3 +193,21 @@ export const fetchPositionIdsForOrderIds = (orderIds: string[]) => gql`
   }
 }
 `;
+
+export const fectchCollateralForOrderUsingAccountId = (accountId: string | string[]) => gql`
```
Comment (sudeepb02) [inline src/subgraph/orders/subgraphQueries.ts]: typo

File: test/pyth-tests/pyth.test.ts [hunk 25]
```diff
@@ -1,59 +0,0 @@
-import 'dotenv/config';
```
Comment (sudeepb02) [inline test/pyth-tests/pyth.test.ts]: Pyth is still relevant with the new integration. Please uncomment and test everything related to Pyth

File: test/relayers/pimlico.test.ts [hunk 26]
```diff
@@ -1,70 +1,73 @@
-import { Chain } from '@parifi/references';
```
Comment (sudeepb02) [inline test/relayers/pimlico.test.ts:1]: Pimlico is still relevant with the new integration. Please uncomment and test everything related to it

File: test/subgraph-tests/accounts.test.ts [hunk 27]
```diff
@@ -1,61 +1,12 @@
 import { getParifiSdkInstanceForTesting } from '..';
```
Comment (sudeepb02) [inline test/subgraph-tests/accounts.test.ts:1]: Account => Wallet
SnxAccount is a different term. Please do not mix them up

File: test/subgraph-tests/index.test.ts [hunk 28]
```diff
@@ -1,29 +0,0 @@
-import { Chain } from '@parifi/references';
```
Comment (sudeepb02) [inline test/subgraph-tests/index.test.ts]: Add this file again. This function is still relevant

**Review by sudeepb02 — CHANGES_REQUESTED** [review]
@iamamitkumar0512 @Akhilleshgoswami Please work on the requested changes

File: src/common/subgraphMapper.ts (hunk 10 above)
Comment (iamamitkumar0512) [reply src/common/subgraphMapper.ts:40]: No I dont think so

File: src/common/subgraphMapper.ts (hunk 12 above)
Comment (iamamitkumar0512) [reply src/common/subgraphMapper.ts:136]: yes

File: src/common/subgraphMapper.ts (hunk 13 above)
Comment (iamamitkumar0512) [reply src/common/subgraphMapper.ts]: sure

File: src/subgraph/markets/subgraphQueries.ts (hunk 5 above)
Comment (Akhilleshgoswami) [reply src/subgraph/markets/subgraphQueries.ts:26]: so I have only written test case for the order, positions, and account as we were using the market from the synthetix  parifi SDK , I didn't change the code here

File: package.json (hunk 9 above)
Comment (iamamitkumar0512) [reply package.json]: Done

File: src/common/subgraphMapper.ts (hunk 10 above)
Comment (iamamitkumar0512) [reply src/common/subgraphMapper.ts:40]: Done

File: src/common/subgraphMapper.ts (hunk 11 above)
Comment (iamamitkumar0512) [reply src/common/subgraphMapper.ts]: Done

File: src/interfaces/sdkTypes.ts (hunk 14 above)
Comment (iamamitkumar0512) [reply src/interfaces/sdkTypes.ts]: Done

File: src/common/subgraphMapper.ts (hunk 13 above)
Comment (iamamitkumar0512) [reply src/common/subgraphMapper.ts]: done

File: src/interfaces/subgraphTypes.ts (hunk 15 above)
Comment (iamamitkumar0512) [reply src/interfaces/subgraphTypes.ts:161]: Done

File: src/subgraph/orders/index.ts (hunk 22 above)
Comment (iamamitkumar0512) [reply src/subgraph/orders/index.ts:30]: yes

File: src/subgraph/orders/index.ts (hunk 23 above)
Comment (iamamitkumar0512) [reply src/subgraph/orders/index.ts]: Done

**Review by sudeepb02 — APPROVED** [review]

**Pull request merged**